realm/realm-library/src/main/cpp/io_realm_internal_objectstore_OsMongoCollection.cpp

```diff
@@ -69,6 +70,22 @@ static std::function<jobject(JNIEnv*, std::vector<ObjectId>)> collection_mapper_
     return arr;
 };
 
+static std::function<jobject(JNIEnv*, RemoteMongoCollection::RemoteUpdateResult)> collection_mapper_update = [](JNIEnv* env, RemoteMongoCollection::RemoteUpdateResult result) {
+    Bson matched_count(result.matched_count);
+    Bson modified_count(result.modified_count);
+    Bson upserted_value;
+    if (result.upserted_id) {
+        upserted_value = new Bson(result.upserted_id.value());
+    }
+    // FIXME: maybe not the most efficient way. Suggestions?
+    BsonArray result_values { matched_count, modified_count, upserted_value };
```

Comment:
I made the assumption returning a BsonArray to Java was the least annoying option given my ignorance on how to smartly work with the JNI. This solution isn't my cup of tea, as it relies on knowing what's in every array position in Java. I'm all open to change this.

Reply:
In any case, the OS collection method I'm using returns an array of Bsons. I tried to create a plain Bson using our protocol to deliver the result to Java and it still gets delivered as a BsonArray, so I think this solution is acceptable. It's just I'm not all too happy about using indices.